### PR TITLE
fix: Default images on success dialog even before signing it generated previews well

### LINKED ISSUE #9699: Listing at drop page shows default picture
Is it possible to show the currently minted item as on the previous page of the dialogue? 
I've seen it, and on the listing, I see the default.

image

### LINKED ISSUE #9921: Default images on success dialog even before signing it generated previews well
On minting dialog it shows ok previews well

on success it was default 
any idea?
image
edit

sometimes is ok

image

### LINKED ISSUE #9978: Listing Success modal doesn't show correct thumbnail
Listing success modal: 

image

NFT:

https://canary.kodadot.xyz/ahp/gallery/112-2404263931

### PR DESCRIPTION
**Pull request by hassnian**
## PR Type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

## Context

- [x] Closes #9921
- [x] Closes #9699
- [x] Closes #9978 (added by prury)

## Needs QA check

- @kodadot/qa-guild please review

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=16faLfsywwNATaEfbH2ah75dn6ZmctQWpMS5G4KFhbmj5hnD)

## Screenshot 📸

- [x] My fix has changed **something** on UI; 

| before signing    | sucess |
| -------- | ------- |
|    |     |

### CODE REVIEW

File: components/common/listingCart/SuccessfulListingBody.vue [hunk 1]
```diff
@@ -28,10 +28,11 @@ const { accountId } = useAuth()
 const mediaItems = computed<ItemMedia[]>(() =>
   props.items.map((item) => ({
     id: item.id,
-    image: item.meta?.image as string,
+    image: (item.animationUrl?.url || item.meta?.image) as string,
```
Comment (vikiival) [inline components/common/listingCart/SuccessfulListingBody.vue:31]: animationUrl at this point is full gen art no ?

Comment (hassnian) [reply components/common/listingCart/SuccessfulListingBody.vue:31]: yes, `animationUrl` is not a static image

https://image-beta.w.kodadot.xyz/ipfs/bafybeicdv7iu5racra3vsvg2k2uvp2jxvb534h67f6lqdwidhmvzg2tjou/?hash=0xed3dcf74fda65f4797e006c2101350fd7703d253fae8c26c932f6b10f96d46e1 

side note: I was thinking if we should use static images any time we have more than x amount(3 maybe) of items , 
that would improve performance , we've see how some drops do consume a lot of resources.

Comment (vikiival) [reply components/common/listingCart/SuccessfulListingBody.vue:31]: Yes, thats why this is not a correct fix :)

**Review by vikiival — DISMISSED** [review]
Preview should be an image not a full genart in a thumbnail

File: components/common/listingCart/SuccessfulListingBody.vue (hunk 1 above)
Comment (hassnian) [reply components/common/listingCart/SuccessfulListingBody.vue]: fixed

**Comment by hassnian** [comment]
> Preview should be an image not a full genart in a thumbnail

now we use images for the **minting success** step and **listing modal**  

https://github.com/kodadot/nft-gallery/assets/44554284/f0abf4e7-d85d-4622-9517-814285125386

CleanShot 2024-04-16 at 11 08 49@2x

File: components/collection/drop/modal/paid/MintOverview.vue [hunk 2]
```diff
@@ -26,7 +26,7 @@
               :src="
                 toMintNft.imageDataPayload
                   ? toMintNft.imageDataPayload.image
-                  : sanitizeIpfsUrl(toMintNft.image)
+                  : sanitizeIpfsUrl(toMintNft.animationUrl)
```
Comment (vikiival) [inline components/collection/drop/modal/paid/MintOverview.vue:29]: so are youu still using full drop right?

Comment (hassnian) [reply components/collection/drop/modal/paid/MintOverview.vue:29]: This is needed to load once the iframe and capture the image data payload , after that only images are used

**Comment by vikiival** [comment]
cc @daiagi 👀

**Comment by prury** [comment]
:white_check_mark: 
image

**Comment by hassnian** [comment]
after merge

https://github.com/kodadot/nft-gallery/assets/44554284/c1e8b4d7-2c34-47bd-91f7-a6b45d543dd0

**Comment by prury** [comment]
@vikiival can i have your approval again
auto merge was enabled but it won't go:

image

**Review by prury — APPROVED** [review]

**Comment by hassnian** [comment]
> @vikiival can i have your approval again auto merge was enabled but it won't go:
> 
> image

@vikiival we need your approval

**Pull request merged**